src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const objArr = [];
+  const newState = { ...state };
+
+  for (const a of actions) {
+    switch (a.type) {
+      case 'addProperties':
+        Object.assign(newState, a.extraData);
+        objArr.push({ ...newState });
```

Comment:
[DONT REPEAT YOURSELF] - Pushing a copy of the object to the array is repeated in each case. It's better to do this once after the switch statement to avoid code repetition.